source/material_model/visco_plastic.cc

```diff
@@ -208,7 +208,51 @@ namespace aspect
             {
               // Use thermal conductivity values specified in the parameter file, if this
               // option was selected.
-              out.thermal_conductivities[i] = MaterialUtilities::average_value (volume_fractions, thermal_conductivities, MaterialUtilities::arithmetic);
```

Comment:
You can keep this line and then only multiple `out.thermal_conductivities` below. Then you also don't need the `else` branch.